lms/views/api/blackboard/files.py

```diff
@@ -37,7 +46,15 @@ def list_files(self):
     @view_config(request_method="GET", route_name="blackboard_api.files.via_url")
     def via_url(self):
         """Return the Via URL for annotating the given Blackboard file."""
-        self.request.find_service(name="oauth2_token").get()
+        # Get the user's access token from the DB.
+        # We're not actually going to *use* the access token, since this view
+        # just returns hard-coded URLs. But we want to *get* the access token
+        # so that we can raise an exception if it's missing and trigger the
+        # authentication flow.
+        try:
+            self.request.find_service(name="oauth2_token").get()
+        except NoOAuth2Token as err:
+            raise ProxyAPIAccessTokenError() from err
```

Comment:
Looks like some duplication here. Would this class benefit from a `_oauth2_token()` method?

Reply:
Maybe, but none of this code is going to be here in the end anyway. Looking up access tokens in the DB and communicating with the Blackboard API will move into a `BlackboardAPIService`. These are just fake placeholder views, and they just need to raise this exception to trigger the authorization flow.

Reply:
Fair enough. If it's a stepping stone.